docker/setup-scripts/zts-auto-config.sh

```diff
@@ -0,0 +1,135 @@
+
+echo '2. get a server certificate for ZTS' | colored_cat g
+echo ${ZTS_CERT_KEY_PATH} | colored_cat y
+echo ${ZTS_CERT_PATH} | colored_cat y
+
+echo '3. create ZTS key pairs for signing Athenz token' | colored_cat g
+openssl genrsa -out "${ZTS_PRIVATE_KEY_PATH}" 4096 2> /dev/null
+openssl rsa -pubout -in "${ZTS_PRIVATE_KEY_PATH}" -out "${ZTS_PUBLIC_KEY_PATH}"
+
+echo '4. create trust store containing all the trusted CAs' | colored_cat g
+rm -f "${ZTS_TRUSTSTORE_PATH}"
+
+CERT_ALIAS='athenz_ca'
+openssl x509 -outform pem -in "${ATHENZ_CA_PATH}" | keytool -importcert -noprompt \
+    -keystore "${ZTS_TRUSTSTORE_PATH}" -storepass "${ZTS_TRUSTSTORE_PASS}" \
+    -storetype JKS -alias "${CERT_ALIAS}"
+
+CERT_ALIAS='user_ca'
+openssl x509 -outform pem -in "${USER_CA_PATH}" | keytool -importcert -noprompt \
+    -keystore "${ZTS_TRUSTSTORE_PATH}" -storepass "${ZTS_TRUSTSTORE_PASS}" \
+    -storetype JKS -alias "${CERT_ALIAS}"
+
+CERT_ALIAS='service_ca'
+openssl x509 -outform pem -in "${SERVICE_CA_PATH}" | keytool -importcert -noprompt \
+    -keystore "${ZTS_TRUSTSTORE_PATH}" -storepass "${ZTS_TRUSTSTORE_PASS}" \
+    -storetype JKS -alias "${CERT_ALIAS}"
+
+echo '5. create key store containing the ZTS server certificate' | colored_cat g
+openssl pkcs12 -export -noiter -nomaciter \
+    -out "${ZTS_KEYSTORE_PATH}" -passout "pass:${ZTS_KEYSTORE_PASS}" \
+    -in "${ZTS_CERT_PATH}" -inkey "${ZTS_CERT_KEY_PATH}"
+
+echo '6. set up for certificate signing' | colored_cat g
+echo 'We will just use KeyStoreCertSigner !!!' | colored_cat y
+# create key store
+openssl pkcs12 -export -noiter -nomaciter \
+    -out "${ZTS_SIGNER_KEYSTORE_PATH}" -passout "pass:${ZTS_SIGNER_KEYSTORE_PASS}" \
+    -in "${ZTS_SIGNER_CERT_PATH}" -inkey "${ZTS_SIGNER_CERT_KEY_PATH}"
+
+# create trust store (only used by HttpCertSigner)
+rm -f "${ZTS_SIGNER_TRUSTSTORE_PATH}"
+CERT_ALIAS='athenz_ca'
+openssl x509 -outform pem -in "${ATHENZ_CA_PATH}" | keytool -importcert -noprompt \
+    -keystore "${ZTS_SIGNER_TRUSTSTORE_PATH}" -storepass "${ZTS_SIGNER_TRUSTSTORE_PASS}" \
+    -storetype JKS -alias "${CERT_ALIAS}"
+
+echo '7. set up for ZMS connection' | colored_cat g
+# create key store
+openssl pkcs12 -export -noiter -nomaciter \
+    -out "${ZMS_CLIENT_KEYSTORE_PATH}" -passout "pass:${ZMS_CLIENT_KEYSTORE_PASS}" \
+    -in "${ZMS_CLIENT_CERT_PATH}" -inkey "${ZMS_CLIENT_CERT_KEY_PATH}"
+
+# create trust store for verifying the ZMS server certificate
+rm -f "${ZMS_CLIENT_TRUSTSTORE_PATH}"
+CERT_ALIAS='athenz_ca'
+openssl x509 -outform pem -in "${ATHENZ_CA_PATH}" | keytool -importcert -noprompt \
+    -keystore "${ZMS_CLIENT_TRUSTSTORE_PATH}" -storepass "${ZMS_CLIENT_TRUSTSTORE_PASS}" \
+    -storetype JKS -alias "${CERT_ALIAS}"
+
+echo '8. summary' | colored_cat g
+tree "${CA_DIR}"
+tree "${PROD_ZTS_DIR}"
+tree "${ZTS_DIR}"
+
+echo '9. register ZTS service to Athenz' | colored_cat g
+ENCODED_ZTS_PUBLIC_KEY=`base64 -w 0 "${ZTS_PUBLIC_KEY_PATH}" | tr '\+\=\/' '\.\-\_'`
```

Comment:
Probably we need a comment.

Or, as far as i tried, It seems that both industry standard encoding and ybase64-encoding are accepted as request value, can you use the industry standard? 

https://github.com/yahoo/athenz/blob/master/libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/util/Crypto.java#L345-L351

Reply:
thx, comment added in https://github.com/WindzCUHK/athenz/pull/2/commits/551ea239876b56b320ad81721350f79e879d139f

Are you referring to https://tools.ietf.org/html/rfc4648#page-7?
I think only if the public key contains a value `62` will cause a problem.
Can you help to check that whether you dataset has a value 62?

Reply:
Thank you for reply!

BTW, I'm curious about follows.

```
$ uname -a
Linux athenz-zts-server-cd45b69db-9s8bp 4.19.86-coreos #1 SMP Mon Dec 2 20:13:38 -00 2019 x86_64 Linux

$ base64 /athenz/keys/zts_public.pem | tr -d '\n' | tr '\+\=\/' '\.\-\_'
LS0tLS1 <snip> yBLRVktLS0tLQo_
```

I understand the result is unintended encoding (last letter). But the ZMS seems accept that.

Input for `${ZMS_URL}/zms/v1/domain/sys.auth/service/zts`
`{"name": "sys.auth.zts","publicKeys": [{"id": "0","key": "LS0tLS1 <<snip>> yBLRVktLS0tLQo_"}]}`

The ZMS returns HTTP 204 response.

Output from `${ZMS_URL}/zms/v1/domain/sys.auth/service/zts`
`{"name":"sys.auth.zts","publicKeys":[{"key":"LS0tLS1 <<snip>> yBLRVktLS0tLQo.","id":"0"}],"modified":"2020-02-17T07:52:48.210Z"}`

Is this normal behavior for ZMS?

Reply:
> It seems that both industry standard encoding and ybase64-encoding are accepted as request value, can you use the industry standard?

As other Athenz components (like ZTS, ZPE) is also using ybase64, we need to use ybase64 as a standard in Athenz. Using the industry standard, may have some compatibility issue.
https://github.com/yahoo/athenz/blob/bbfba4bf41b5f0f585eac7ba681300928fb5859e/servers/zts/src/main/java/com/yahoo/athenz/zts/store/DataStore.java#L166

> BTW, I'm curious about follows.

Thank you very much for pointing out. Can you post an issue in original repo?
https://github.com/yahoo/athenz/issues